server/controllers/lecture-controller.js

```diff
@@ -405,6 +406,106 @@ exports.deleteLecture = async (req, res) => {
   }
 };
 
+// Record the attendances of a lecture  
+exports.recordAttendances = async (req, res) => {
+  const userId = req.user.id;
+  const lectureId = req.params.lectureid;
+  const presentStudents = req.body;
+  let queries;
+
+  try {
+    if(!presentStudents.length) {
+      throw {
+        msg: `Invalid body: a not-empty array containing the student ids is expected`,
+        status: 422,
+      };
+    }
+    const [lecture] = await knex
+      .select(
+        { id: "l.id" },
+        { start: "l.start" },
+        { lecturer: "l.lecturer" },
+        { status: "l.status" }
+      )
+      .from({ l: "lecture" })
+      .where("l.id", lectureId);
+
+    if(!lecture) {
+      throw {
+        msg: `The lecture doesn't exist`,
+        status: 404,
+      };
+    } else if(lecture.lecturer != userId) {
+      throw {
+        msg: `Only the teacher of this course can record attendances`,
+        status: 401,
+      };
+    } else if(lecture.status != "presence") {
+      throw {
+        msg: `Cannot record attendances of remote lecture`,
+        status: 400,
+      };
+    } else if( !moment().isSame(moment(lecture.start, "YYYY-MM-DD HH:mm:ss"), "day") ) {
```

Comment:
the lesson should not be selected when it is today but not yet started

Reply:
Resolved